src/App.tsx

```diff
@@ -10,33 +12,23 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTab, setSelectedTab] = useState(tabs[0]);
+
+  const onTabSelected = (tab: Tab) => {
+    setSelectedTab(tab);
+  };
```

Comment:
Misha did mean to move all body into function like

```suggestion
  const onTabSelected = (tab: Tab) => {
    if (tab.id !== selectedTab.id) {
      setSelectedTab(tab);
    }
  };
```